cryptography/src/bls12381/primitives/group.rs

```diff
@@ -1045,4 +1081,196 @@ mod tests {
+        assert_eq!(a == c, a.cmp(&c) == std::cmp::Ordering::Equal);
+
+        // Test that ordering is deterministic
+        let ord1 = a.cmp(&b);
+        let ord2 = a.cmp(&b);
+        assert_eq!(ord1, ord2);
+
+        // Test antisymmetry: if a < b, then b > a
+        match a.cmp(&b) {
+            std::cmp::Ordering::Less => assert_eq!(b.cmp(&a), std::cmp::Ordering::Greater),
+            std::cmp::Ordering::Greater => assert_eq!(b.cmp(&a), std::cmp::Ordering::Less),
+            std::cmp::Ordering::Equal => assert_eq!(b.cmp(&a), std::cmp::Ordering::Equal),
+        }
+    }
+
+    #[test]
+    fn test_g2_ord() {
+        let mut rng = thread_rng();
+        let mut a = G2::one();
+        a.mul(&Scalar::rand(&mut rng));
+        let mut b = G2::one();
+        b.mul(&Scalar::rand(&mut rng));
+        let c = a;
+
+        // Test reflexivity: a == a
+        assert_eq!(a.cmp(&c), std::cmp::Ordering::Equal);
+        assert_eq!(a.partial_cmp(&c), Some(std::cmp::Ordering::Equal));
+
+        // Test consistency with Eq
+        assert_eq!(a == c, a.cmp(&c) == std::cmp::Ordering::Equal);
+
+        // Test that ordering is deterministic
+        let ord1 = a.cmp(&b);
+        let ord2 = a.cmp(&b);
+        assert_eq!(ord1, ord2);
+
+        // Test antisymmetry: if a < b, then b > a
+        match a.cmp(&b) {
+            std::cmp::Ordering::Less => assert_eq!(b.cmp(&a), std::cmp::Ordering::Greater),
+            std::cmp::Ordering::Greater => assert_eq!(b.cmp(&a), std::cmp::Ordering::Less),
+            std::cmp::Ordering::Equal => assert_eq!(b.cmp(&a), std::cmp::Ordering::Equal),
+        }
+    }
+
+    #[test]
+    fn test_hash_ord_consistency() {
+        use std::collections::hash_map::DefaultHasher;
+        use std::hash::{Hash, Hasher};
+
+        let mut rng = thread_rng();
+
+        // Test Scalar
+        let s1 = Scalar::rand(&mut rng);
+        let s2 = s1.clone();
+        assert_eq!(s1.cmp(&s2), std::cmp::Ordering::Equal);
+
+        let mut h1 = DefaultHasher::new();
+        let mut h2 = DefaultHasher::new();
+        s1.hash(&mut h1);
+        s2.hash(&mut h2);
+        assert_eq!(h1.finish(), h2.finish());
+
+        // Test G1
+        let mut g1_1 = G1::one();
+        g1_1.mul(&Scalar::rand(&mut rng));
+        let g1_2 = g1_1;
+        assert_eq!(g1_1.cmp(&g1_2), std::cmp::Ordering::Equal);
+
+        let mut h1 = DefaultHasher::new();
+        let mut h2 = DefaultHasher::new();
+        g1_1.hash(&mut h1);
+        g1_2.hash(&mut h2);
+        assert_eq!(h1.finish(), h2.finish());
+
+        // Test G2
+        let mut g2_1 = G2::one();
+        g2_1.mul(&Scalar::rand(&mut rng));
+        let g2_2 = g2_1;
+        assert_eq!(g2_1.cmp(&g2_2), std::cmp::Ordering::Equal);
+
+        let mut h1 = DefaultHasher::new();
+        let mut h2 = DefaultHasher::new();
+        g2_1.hash(&mut h1);
+        g2_2.hash(&mut h2);
+        assert_eq!(h1.finish(), h2.finish());
+    }
+
+    #[test]
```

Comment:
I'm torn on whether this test is necessary. If we implement the required traits, I don't think we need to test that it "works" here.

If we are concerned with some sort of overwrite issue (where the hashes are the same), I think we should test that explicitly rather than via this approach.